src/Maestro/SubscriptionActorService/PullRequestActor.cs

```diff
@@ -1075,6 +1079,49 @@ private async Task<RepositoryBranchUpdate> GetRepositoryBranchUpdate()
 
         private static string GetNewBranchName(string targetBranch) => $"darc-{targetBranch}-{Guid.NewGuid()}";
 
+        /// <summary>
+        ///     Given a set of updates, replace the `from` version of every dependency update with the corresponding version
+        ///     from the target branch 
+        /// </summary>
+        /// <param name="darcRemote">Darc client used to fetch target branch dependencies.</param>
+        /// <param name="targetRepository">Target repository to fetch the dependencies from.</param>
+        /// <param name="targetBranch">Target branch to fetch the dependencies from.</param>
+        /// <param name="targetRepositoryUpdates">Incoming updates to the repository</param>
+        /// <returns>
+        ///     Asset update and the corresponding list of altered dependencies
+        /// </returns>
+        /// <remarks>
+        ///     This method is intended for use in situations where we want to keep the information about the original dependency
+        ///     version, such as when updating PR descriptions.
+        /// </remarks>
+        private static async Task<List<(UpdateAssetsParameters update, List<DependencyUpdate> deps)>> ReplaceFromDependency(
+            IRemote darcRemote,
+            string targetRepository,
+            string targetBranch,
+            TargetRepoDependencyUpdate targetRepositoryUpdates
+            )
+        {
+            List<DependencyDetail> targetBranchDeps = (await darcRemote.GetDependenciesAsync(targetRepository, targetBranch)).ToList();
+
+            List<(UpdateAssetsParameters update, List<DependencyUpdate> deps)> alteredUpdates = [];
```

Comment:
Technically you could have kept the select (and do `.Select(update => new() ...`) still but I think this is maybe even more readable.